### PR TITLE
[CWS] Restore ability to trigger dump over gRPC

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

Restore the ability to trigger an activity dump using the gRPC API.

### Motivation

PR https://github.com/DataDog/datadog-agent/pull/27030 removed complete support for on-demand activity dump generation.

### CODE REVIEW

File: pkg/security/proto/api/api.pb.go [hunk 1]
```diff
@@ -1584,6 +1584,85 @@ func (x *StorageRequestParams) GetRemoteStorageCompression() bool {
 	return false
 }
 
+type ActivityDumpParams struct {
+	state         protoimpl.MessageState
+	sizeCache     protoimpl.SizeCache
+	unknownFields protoimpl.UnknownFields
+
+	Comm              string                `protobuf:"bytes,1,opt,name=Comm,proto3" json:"Comm,omitempty"`
```
Comment (spikat) [inline pkg/security/proto/api/api.pb.go:1592]: Could you please regen the API files, comm are still present

File: cmd/system-probe/subcommands/runtime/activity_dump.go [hunk 2]
```diff
@@ -117,11 +119,84 @@ func generateCommands(globalParams *command.GlobalParams) []*cobra.Command {
...
+
+	activityDumpGenerateDumpCmd := &cobra.Command{
+		Use:   "dump",
+		Short: "generate an activity dump",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return fxutil.OneShot(generateActivityDump,
+				fx.Supply(cliParams),
+				fx.Supply(core.BundleParams{
+					ConfigParams: config.NewAgentParams("", config.WithConfigMissingOK(true)),
+					SecretParams: secrets.NewEnabledParams(),
```
Comment (brycekahle) [inline cmd/system-probe/subcommands/runtime/activity_dump.go:141]: should this be `NewDisabledParams`?

**Review by spikat — APPROVED** [review]

File: cmd/system-probe/subcommands/runtime/activity_dump.go (hunk 2 above)
Comment (paulcacheux) [reply cmd/system-probe/subcommands/runtime/activity_dump.go]: yes, I updated it in the latest version

**Review by brycekahle — APPROVED** [review]

**Pull request merged**